unit_test/fuzzing/fuzzing.sh

```diff
@@ -0,0 +1,157 @@
+    pushd $libspdm_path
+    latest_hash=`git log --pretty="%h" -1`
+    export fuzzing_out=$libspdm_path/unit_test/fuzzing/out_$1_$latest_hash
+    export build_fuzzing=build_fuzz_$1_$latest_hash
+fi
+
+if [ ! -d "$fuzzing_out" ];then
+    mkdir $fuzzing_out
+fi
+
+for i in $fuzzing_out/*;do
+    if [[ ! -d $i/crashes ]] && [[ ! -d $i/hangs ]];then
+        continue
+    fi
+
+    if [[ "`ls -A $i/crashes`" != "" ]];then
+        echo -e "\033[31m There are some crashes \033[0m"
+        echo -e "\033[31m Path in $i/crashes \033[0m"
+        exit
+    fi
+
+    if [[ "`ls -A $i/hangs`" != "" ]];then
+        echo -e "\033[31m There are some hangs \033[0m"
+        echo -e "\033[31m Path in $i/hangs \033[0m"
+        exit
+    fi
+done
+
+rm -rf $fuzzing_out/*
+
+if [[ "core" != `cat /proc/sys/kernel/core_pattern` ]];then
+    # Here 'test' is the sudo password, replace it with yours.
+    echo 'test' | sudo -S bash -c 'echo core >/proc/sys/kernel/core_pattern'
+    pushd /sys/devices/system/cpu/
+    echo 'test' | sudo -S bash -c 'echo performance | tee cpu*/cpufreq/scaling_governor'
+    popd
+fi
+
+if [ -d "$build_fuzzing" ];then
+    rm -rf $build_fuzzing
+fi
+
+mkdir $build_fuzzing
+pushd $build_fuzzing
+
+cmake -DARCH=x64 -DTOOLCHAIN=AFL -DTARGET=Release -DCRYPTO=$1 -DGCOV=$2 ..
+make copy_sample_key
+make
+pushd bin
+
+cmds=(
+test_spdm_requester_get_version
+test_spdm_requester_get_capabilities
+test_spdm_requester_negotiate_algorithms
+test_spdm_requester_get_digests
+test_spdm_requester_get_certificate
+test_spdm_requester_challenge
+test_spdm_requester_get_measurements
+test_spdm_requester_key_exchange
+test_spdm_requester_finish
+test_spdm_requester_psk_exchange
+test_spdm_requester_psk_finish
+test_spdm_requester_heartbeat
+test_spdm_requester_key_update
+test_spdm_requester_end_session
+test_spdm_responder_version
+test_spdm_responder_capabilities
+test_spdm_responder_algorithms
+test_spdm_responder_digests
+test_spdm_responder_certificate
+test_spdm_responder_challenge_auth
+test_spdm_responder_measurements
+test_spdm_responder_key_exchange
+test_spdm_responder_finish_rsp
+test_spdm_responder_psk_exchange_rsp
+test_spdm_responder_psk_finish_rsp
+test_spdm_responder_heartbeat_ack
+test_spdm_responder_key_update
+test_spdm_responder_end_session
+)
+for ((i=0;i<${#cmds[*]};i++))
+do
+    echo ${cmds[$i]}
+    screen -ls | grep ${cmds[$i]}
+    if [[ $? -ne 0 ]]
+    then
+    screen -dmS ${cmds[$i]}
+    fi
+    screen -S ${cmds[$i]} -p 0 -X stuff "afl-fuzz -i $fuzzing_seeds/${cmds[$i]} -o $fuzzing_out/${cmds[$i]} ./${cmds[$i]} @@"
+    screen -S ${cmds[$i]} -p 0 -X stuff $'\n'
+    sleep $duration
+    screen -S ${cmds[$i]} -X quit
+    sleep 5
+done
+
+if [[ $2 = "ON" ]]; then
+    cd $fuzzing_out
+    mkdir coverage_log
+    cd coverage_log
+    lcov --capture --directory $libspdm_path --output-file coverage.info
+    genhtml coverage.info --output-directory .
+fi
+
+cd $script_path
+python3 ./summary.py $fuzzing_out
```

Comment:
Do we must use python? Can we just use normal shell script?
If have to, then we need document python requirement, e.g python3 version.

Reply:
Python is not necessary, and we can just use the normal shell script. I have updated.

Reply:
Thank you!